server/service/src/processors/transfer/shipment/common.rs

```diff
@@ -0,0 +1,74 @@
+use repository::Invoice;
+use repository::{InvoiceLineRow, InvoiceLineRowType, RepositoryError, StorageConnection};
+use util::uuid::uuid;
+
+use crate::invoice::common::get_lines_for_invoice;
+
+pub(crate) fn regenerate_inbound_shipment_lines(
```

Comment:
Just an idea but from what I can see `regenerate_inbound_shipment_lines` is used in two places, once in `update_inbound_shipment_lines` and once in `create_inbound_shipment_lines`.

Only one of which needs the ids to delete.
What about just calling this `generate_inbound_shipment_lines` and moving the line_ids_to_delete logic into the `update_inbound_shipment_lines`?

That would feel more consistent to me...

Reply:
:nit
for me,  this fn doesn't regenerate lines as it doesn't persist the new ones or delete the old - I see it as `generate_inbound_shipment_lines` though it is doing more than that, since it is generating the list of lines to be deleted as well as generating the lines to be created.

following on from JB's comment, I'd have a `generate_lines` method and a `select_lines_to_delete`  method

Reply:
🙏 great idea, smudging multiple operations into one is 👎 , unless of course result of both operations is always desired: d392e4a7bfcc5c1d726464bff2ea9ebfea9fa82d